src/DripsHub.sol

```diff
@@ -355,26 +362,26 @@ abstract contract DripsHub {
                 );
                 split += splitsAmt;
                 address splitsReceiver = currReceivers[i].receiver;
-                splitsStates[calcUserId(splitsReceiver)].collectables[DUMMY_ASSET] += splitsAmt;
-                emit Split(user, splitsReceiver, DUMMY_ASSET, splitsAmt);
+                splitsStates[calcUserId(splitsReceiver)].collectables[assetId] += splitsAmt;
+                emit Split(user, splitsReceiver, assetId, splitsAmt);
             }
             collected -= split;
         }
-        emit Collected(user, DUMMY_ASSET, collected, split);
+        emit Collected(user, assetId, collected, split);
     }
 
     /// @notice Collects and clears user's cycles
     /// @param user The user
+    /// @param assetId The used asset ID
     /// @param count The number of flushed cycles.
     /// @return collectedAmt The collected amount
-    function _flushCyclesInternal(address user, uint64 count)
-        internal
-        returns (uint128 collectedAmt)
-    {
+    function _flushCyclesInternal(
+        address user,
```

Comment:
I think it might be worth consider changing the pattern a bit. We could already pass here the `bytes32 userId` instead of the address.

We could calculate the userId on that layer where we deal with `msg.sender`, etc.

I think this would be a bit cleaner. Then the lower layer isn't even aware that a userId is calculated from an address.

Reply:
Maybe you are planning to do this anyway as part of the `userOrAccount` refactoring.

Reply:
I agree, that's the goal, but let's do one thing at a time, this PR is already almost 700 line changes.